### PR TITLE
Implement claim command.

### LINKED ISSUE #6: Ability to claim ownership of tags from users that have left
This issue will add a new `/tags claim <tag: str>` command. When it is used, it will check if a tag by that name is owned by a user who is not in the server. If so, the tag ownership will be transferred to the user of the command. Optionally create a helper function named `claim_tag` which will be used by both `/tags claim` and `/tags transfer` to reduce redundancy.

Part of #1.

### PR DESCRIPTION
**Pull request by bsrecker**
Claims an unclaimed tag. An unclaimed tag is a tag with no owner because they have left the server. #6

### CODE REVIEW

File: leaf/extensions/tags.py [hunk 1]
```diff
@@ -402,6 +402,55 @@ async def transfer_tag(
...
+        async with self.bot.database.transaction():
+            tag_record = await self.bot.database.fetchrow(
+                "SELECT * FROM Tags WHERE name = $1 AND guild_id = $2 AND deleted = FALSE;",
+                tag,
+                interaction.guild.id,
+            )
+
+            if tag_record:
+                try:
+                    member = await interaction.guild.fetch_member(
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Shouldn't we be using `Guild.get_member` instead to get the member from cache?

File: leaf/extensions/tags.py [hunk 2]
```diff
@@ -402,6 +402,55 @@ async def transfer_tag(
...
+                        tag_record["owner_id"]
+                    )
+                    if member is not None:
+                        await interaction.response.send_message(
+                            embed=discord.Embed(
+                                description=f"Owner of the tag **{tag}** is still present in the server.",
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Maybe include "the" before it? Also, to be consistent we should use quotes around the tag name. `f"The owner of the tag "{tag}" is still present in the server."`

File: leaf/extensions/tags.py [hunk 3]
```diff
@@ -402,6 +402,55 @@ async def transfer_tag(
...
+                        )
+                except discord.NotFound:
+                    await self.bot.database.execute(
+                        "UPDATE tags SET owner_id = $1 WHERE name = $2",
+                        interaction.user.id,
+                        tag,
+                    )
+
+                    await interaction.response.send_message(
+                        interaction.user.mention,
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Don't ping the user here because they are already in the channel.

File: leaf/extensions/tags.py [hunk 4]
```diff
@@ -402,6 +402,55 @@ async def transfer_tag(
                     )
                 )
 
+    @app_commands.describe(tag="The tag you wish to claim.")
+    @app_commands.command(
+        name="claim",
+        description="Claims an unclaimed tag. An unclaimed tag is a tag with no owner "
+        "because they have left the server.",
+    )
+    async def claim_tag(self, interaction: discord.Interaction, tag: str) -> None:
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Should we have a silent parameter here?

File: leaf/extensions/tags.py [hunk 5]
```diff
@@ -402,6 +402,55 @@ async def transfer_tag(
...
+                        embed=discord.Embed(
+                            description=f"The tag has successfully been claimed by {interaction.user.mention}.",
+                            color=discord.Color.dark_embed(),
+                        ),
+                    )
+            else:
+                await interaction.response.send_message(
+                    interaction.user.mention,
+                    embed=discord.Embed(
+                        description=f"A tag with the name of **{tag}** does not exist",
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: This should be `f"A tag named "{tag}" does not exist."`.

File: leaf/extensions/tags.py [hunk 6]
```diff
@@ -402,6 +402,55 @@ async def transfer_tag(
...
+                    await interaction.response.send_message(
+                        interaction.user.mention,
+                        embed=discord.Embed(
+                            description=f"The tag has successfully been claimed by {interaction.user.mention}.",
+                            color=discord.Color.dark_embed(),
+                        ),
+                    )
+            else:
+                await interaction.response.send_message(
+                    interaction.user.mention,
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Ping is not necessary here again.

**Review by mudkipdev — CHANGES_REQUESTED** [review]

File: leaf/extensions/tags.py (hunk 3 above)
Comment (bsrecker) [reply leaf/extensions/tags.py]: updated

File: leaf/extensions/tags.py (hunk 6 above)
Comment (bsrecker) [reply leaf/extensions/tags.py]: updated.

File: leaf/extensions/tags.py (hunk 5 above)
Comment (bsrecker) [reply leaf/extensions/tags.py]: Updated

File: leaf/extensions/tags.py (hunk 1 above)
Comment (bsrecker) [reply leaf/extensions/tags.py]: Added a function to bot.py to try members:

```
async def try_member(self, id: int, /, *, guild: discord.Guild) -> discord.Member:
    return guild.get_member(id) or await guild.fetch_member(id)
```

File: leaf/extensions/tags.py (hunk 4 above)
Comment (bsrecker) [reply leaf/extensions/tags.py]: Added

File: leaf/extensions/tags.py [hunk 7]
```diff
@@ -436,19 +439,19 @@ async def claim_tag(self, interaction: discord.Interaction, tag: str) -> None:
...
                             color=discord.Color.dark_embed(),
                         ),
+                        ephemeral=silent,
                     )
             else:
                 await interaction.response.send_message(
-                    interaction.user.mention,
                     embed=discord.Embed(
-                        description=f"A tag with the name of **{tag}** does not exist",
+                        description=f"A tag named **{tag}** does not exist",
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: This is still bold instead of quotes.

Comment (bsrecker) [reply leaf/extensions/tags.py]: fixed

**Comment by mudkipdev** [comment]
LGTM. Merging now.

**Pull request merged**